src/vmax.rs

```diff
@@ -1240,22 +1239,52 @@ fn test_get_vmax_json_volume() {
     println!("point: {:#?}", i.into_point(Some("vmax_slo_volume")));
 }
 
+#[test]
+/// Test retrieving one page of volumes from json
+fn test_get_vmax_json_volumes() {
+    use std::fs::File;
+    use std::io::Read;
+
+    let mut f = File::open("tests/vmax/slo_volumes.json").unwrap();
+    let mut buff = String::new();
+    f.read_to_string(&mut buff).unwrap();
+
+    let data: Value = serde_json::from_str(&buff).unwrap();
+
+    // grab the list from what was returned before calling the iterator
+    let all_volume_ids = match data["resultList"]["result"].as_array() {
+        Some(v) => v
+            .iter()
+            .map(|val| {
+                val.as_object()
+                    .unwrap()
+                    .values()
+                    .map(|v_i| v_i.as_str().unwrap().to_string())
+                    .collect::<String>()
+            })
+            .collect(),
+        None => vec![],
+    };
+
+    println!("Volume IDs: {:?}", all_volume_ids);
+}
+
 /// Returns a list of volume IDs for this array
 // TODO: combine this with the other getters and generalize along with other metrics
 pub fn get_all_slo_volumes(
     client: &reqwest::Client,
     config: &VmaxConfig,
     symmetrixid: &str,
 ) -> MetricsResult<Vec<String>> {
-    let data: Value = client
-        .get(&format!(
+    let data: Value = super::get(
```

Comment:
haha ok well that only saves 1 line of code but at least it's not duplicate code